### ISSUE TITLE
rgbds 0.4.2-pre segfaults at a `newcharmap name, basename` with a "warning: Invalid allocation size"

### ISSUE THREAD

**Issue by Rangi42** [gbdev/rgbds#586]
With https://github.com/Rangi42/polishedcrystal/commit/6849c2596a71a7ffe59021a246db54720f854523, I create an `rgbds/` subdirectory for the 0.4.2-pre files so `make` will use them instead of the ones in `$PATH`.

```
rgbds/rgbasm -E -Weverything -L -o main.o main.asm
make: *** [Makefile:123: main.o] Segmentation fault
```

Running under `gdb`:

```
Starting program: rgbds/rgbasm -E -Weverything -L -o main.o main.asm
[New Thread 10856.0xab8]
warning: HEAP[rgbasm.exe]:
warning: Invalid allocation size - 77f777f777f77018 (exceeded 7ffffffdefff)
FATAL: main.asm(1) -> constants.asm(5) -> charmap.asm(216):
    Failed to resize charmap: Not enough space
[Inferior 1 (process 10856) exited with code 01]
```

[charmap.asm:216](https://github.com/Rangi42/polishedcrystal/blob/6849c2596a71a7ffe59021a246db54720f854523/charmap.asm#L216) is:

```
	newcharmap default, no_ngrams
```

(The `no_ngrams` charmap had already been started from line 1.)

**Comment by Rangi42** [comment]
The stacktrace from `gdb` when this occurs:

```
#0  resizeCharmap (map=0x80006c020, capacity=64) at src/asm/charmap.c:64
#1  0x0000000100404ec3 in charmap_Add (mapping=0xfffeffac "ND>", value=6 '\006')
    at src/asm/charmap.c:175
#2  0x0000000100402561 in yyparse () at src/asm/asmy.y:752
#3  0x000000010040a859 in main (argc=1, argv=0xffffcc40) at src/asm/main.c:503
```

**Comment by Rangi42** [comment]
Fixed by #593.
